metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py

```diff
@@ -29,6 +27,18 @@ class StructuredLinkableSpecName:
     time_granularity_name: Optional[str] = None
     date_part: Optional[DatePart] = None
 
+    def __init__(  # noqa: D107
+        self,
+        entity_link_names: Tuple[str, ...],
+        element_name: str,
+        time_granularity_name: Optional[str] = None,
+        date_part: Optional[DatePart] = None,
+    ) -> None:
+        self.entity_link_names = tuple([entity_link_name.lower() for entity_link_name in entity_link_names])
+        self.element_name = element_name.lower()
+        self.time_granularity_name = time_granularity_name.lower() if time_granularity_name else None
+        self.date_part = date_part
+
     def __post_init__(self) -> None:
```

Comment:
This is a dataclass construct and will no longer be called. Let's merge the remaining bits into `__init__` and then remove this method.

Reply:
Oooh TIL!! I thought this applied to all Python classes!